tests/components/element-editor/basic-details/basic-details.spec.tsx

```diff
@@ -19,232 +19,155 @@ describe("BasicDetails", () => {
             );
 
-            expect(dropdownItemLabels[0]).toBeInTheDocument();
-            expect(dropdownItemValues[0]).toBeInTheDocument();
-
-            expect(
-                screen.queryByText("Pre-selected value (optional)")
-            ).not.toBeInTheDocument();
-
-            fireEvent.change(dropdownItemLabels[0], {
-                target: { value: "New Label" },
+            fireEvent.change(descriptionField, {
+                target: { value: "description" },
+            });
+            fireEvent.change(placeholderField, {
+                target: { value: "placeholder" },
             });
-            fireEvent.change(dropdownItemValues[0], {
-                target: { value: "New Value" },
+            fireEvent.change(preselectedValueField, {
+                target: { value: "preselectedValue" },
             });
 
-            expect(dropdownItemLabels[0]).toHaveValue("New Label");
-            expect(dropdownItemValues[0]).toHaveValue("New Value");
+            expect(descriptionField).toHaveValue("description");
+            expect(placeholderField).toHaveValue("placeholder");
+            expect(preselectedValueField).toHaveValue("preselectedValue");
 
-            const preselectedValue = await screen.findByText(
-                "Pre-selected value (optional)"
-            );
+            // Change element type to TEXT
+            fireEvent.click(screen.getByTestId("type-field"));
 
-            expect(preselectedValue).toBeInTheDocument();
-        });
-    });
+            const textOption = await screen.findByText("Short text");
+            expect(textOption).toBeInTheDocument();
 
-    describe("submitting the form for dropdown element", () => {
-        it("should not be able to submit the form if there are not at least 2 valid dropdown items", async () => {
-            renderComponent({
-                builderContext: {
-                    focusedElement: MOCK_FOCUSED_DROPDOWN_ELEMENT,
-                    elements: MOCK_ELEMENTS,
-                },
-                formContext: {
-                    elementType: EElementType.DROPDOWN,
-                    defaultValues: {
-                        type: EElementType.DROPDOWN,
-                        dropdownItems: [
-                            { label: "", value: "" },
-                            { label: "", value: "" },
-                        ],
-                    },
-                },
-            });
+            fireEvent.click(textOption);
 
-            const submitButton = screen.getByText("Submit");
-            fireEvent.click(submitButton);
+            // Check that the fields are cleared
+            waitFor(() => {
+                descriptionField = screen.getByTestId("description-field");
+                placeholderField = screen.getByTestId("placeholder-field");
+                preselectedValueField = screen.getByTestId(
+                    "preselected-value-field"
+                );
```

Comment:
not needed, can rely on the initial queries at the beginning of this test

Reply:
By using the initial queries, I don't seem to be getting the updated values after the `useEffect` runs.